web/styles/left_sidebar.css

```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
     }
 }
 
-.narrows_panel {
-    & li a {
-        margin-top: 1px;
+#views-label-container {
+    cursor: pointer;
+    /* Padding of #private_messages_section_header. */
+    padding: 0 10px 1px 6px;
+    /* The margin-left of .private_messages_container. */
+    margin-left: 6px;
+    /* Space between #views-label-container and #left-sidebar-navigation-list in collapsed state. */
+    margin-right: 10px;
+    margin-top: 3px;
+    outline: none;
+}
 
-        &:hover {
+#left-sidebar-navigation-area {
+    & li.top_left_row {
+        position: relative;
+        padding: 2px 10px 1px $far_left_gutter_size;
+
+        & a:hover {
             text-decoration: none;
         }
     }
+
+    &.collapsed {
+        display: flex;
+        align-items: center;
+        /* 2px space between views and direct messages sections when views section is collapsed. */
+        margin-bottom: 2px;
+
+        /* Only show first 5 views in collapsed state. */
+        .top_left_row:nth-child(n + 6) {
+            display: none !important;
+        }
+
+        & #left-sidebar-navigation-list {
+            width: 100%;
+            display: flex;
+            /* 25px margin-right of .subscription_block + 12px margin-right of #stream_filters. */
+            margin-right: 37px;
+            margin-bottom: 0;
+            padding: 0;
+
+            & li {
+                padding: 0;
+                flex-grow: 1;
+
+                .left-sidebar-navigation-label-container {
+                    padding: 0;
+                    justify-content: center;
+
+                    .filter-icon {
+                        padding-top: 4px;
+
+                        & i {
+                            padding-right: 0;
+
+                            &.fa {
+                                font-size: 16px;
+                            }
+
+                            &.zulip-icon-star-filled {
+                                top: 1px;
+                            }
+                        }
+                    }
+
+                    .left-sidebar-navigation-label {
+                        display: none;
+                    }
+                }
+
+                .arrow {
+                    display: none;
+                }
+            }
+        }
+
+        /* CSS to align unread counters in top right corner. */
+        .unread_count {
+            position: absolute;
+            bottom: calc(100% - 4px);
+            left: calc(100% - 4px);
+            z-index: 3;
+            transform: translate(-25%, 25%);
+        }
+
+        /* Until hovered on li, show unread counters as dots. */
+        & li:not(:hover) .unread_count {
```

Comment:
Can we go the other direction and use the `:hover` element on the `.unread_count` above? That will better reflect the state of things in the stylesheet, and not make people's heads ache trying to parse `:not(:hover)`